shuup/simple_supplier/forms.py

```diff
@@ -23,22 +25,34 @@ class StockAdjustmentForm(forms.Form):
             currency_name=get_currency_name(settings.SHUUP_HOME_CURRENCY),
         )
     )
-    delta = forms.DecimalField(label=_("Quantity"))
+    delta = DecimalPlaceField(label=_("Quantity"))
+    decimal_places = forms.IntegerField(widget=forms.HiddenInput())
```

Comment:
Added a hidden field decimal_places to recover the decimal places value for the validation from the POST data.

Reply:
not good. IMO we should configure the field according to the decimal places passed through __init__. then you can create the field accordingly:

```py
class StockAdjustmentForm(forms.Form):
    delta = forms.DecimalField(label=_("Quantity")) # keep as it is

    def __init__(self, sales_unit: Optional[SalesUnit] = None, *args, **kwargs):
       super().__init__(*args, **kwargs)
       if sales_unit:
       self.fields["delta"].decimal_places = sales_unit.decimal_places
```

could this work? please check https://docs.djangoproject.com/en/2.2/ref/forms/fields/#django.forms.DecimalField.decimal_places

Reply:
`sales_unit` is available when we render the form but not when the process it after the user posts the response. We could do it if we do a specific query for retrieving the sales_unit of the product.